lib/recipes-data/src/lib/extract-recipes.ts

```diff
@@ -54,16 +54,51 @@ function determineRecipeUID(recipeIdField: string, canonicalId: string): string
   }
 }
 
+/**
+ * Composer will pass an ADT, in the format {type: "contributor", tagId: string} | {type: "freetext", text: string}.  We need to put the 'contributor' tags into
+ * the "contributors" array and the "freetext" tags into the "byline" array.  We must also handle the migration case, where we still get a raw string passed over.
+ * @param parsedRecipe raw parsed recipe
+ */
+export function handleFreeTextContribs(parsedRecipe: Record<string, unknown>):Record<string, unknown> {
+  const incomingArray = parsedRecipe.contributors as unknown[];
+
+  const contributorTags:string[] = [];
+  const freetexts: string[] = [];
+
+  incomingArray.forEach((entry)=>{
+    if(typeof entry==='string') { //it's an old one, a contrib tag
+      contributorTags.push(entry);
+    } else if(typeof entry==='object') {  //it's an object
+      const contrib = entry as Contributor;
+      switch(contrib.type) {
+        case "contributor":
+          contributorTags.push(contrib.tagId);
+          break;
+        case "freetext":
+          freetexts.push(contrib.text);
+          break;
+        default:
+          break;
+      }
+    }
+  });
+
+  return {...parsedRecipe, contributors: contributorTags, byline: freetexts}
+}
+
 function parseJsonBlob(canonicalId: string, recipeJson: string): RecipeReferenceWithoutChecksum | null {
   try {
-    const recipeData = JSON.parse(recipeJson) as Record<string, unknown>
+    const recipeData = JSON.parse(recipeJson) as Record<string, unknown>;
```

Comment:
You would also need to update the type assertion on this line:

```suggestion
    const recipeData = JSON.parse(recipeJson) as (Record<string, unknown> & {contributors: Array<string | Contributor>});
```